### PR TITLE
Feature/challenge memory management

### PR DESCRIPTION
**Pull request by jonathansheets517**
@merwanehamadi asked me to steal his PR #3425 feature/challenge-memory-management and update the test_memory_management.py file to iterate through 20 text files to test the agent memory as a form of an "A20" Test that it passes. 

He told me it had to be done in 5 hours, hence the brevity of the PR request.

image

### Background

This is an integration test for Auto-GPT that checks the memory management capabilities of an Agent. The agent is given a task ID and a sequence of files containing instructions to navigate through these files. The agent is expected to write its task ID to an output file upon reaching the final instruction.

The test setup involves:

- Importing the necessary libraries and modules.
- Defining the instructions and file contents as strings.
- Defining a test function test_memory_management() with a pytest decorator that requires an API key.
- Creating the files and writing the instructions to the agent's workspace using the provided helper functions.
- Running the agent's interaction loop for 40 steps.
- Catching a SystemExit exception to handle the agent's shutdown.
- Asserting that the output file contains the correct task ID.

The purpose of this test is to ensure that the agent can efficiently manage its memory and perform the required tasks in a limited number of interaction steps. By submitting this PR, it will be contributing a test that verifies the agent's ability to read and follow instructions across multiple files and complete the given task.

### Changes

The only thing I changed from #3425 is the test_memory_management.py file 
I created 20 .txt files for the agent to look through with their file paths and at the end of the test the Agent successfully remembered its task id "2314" and wrote it correctly to the "output.txt" file in its directory.
 
### Documentation

@merwanehamadi documented the code changes well enough to create an A20 level test that passes.

### Test Plan

Test outside of VScode:
image

Test inside VScode:
image

### PR Quality Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [ ] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by holisHsu** [comment]
Cool 👍 

Seems the only change is the commit `e18d279adb0b7a8a6cf0f567f4cbc85cc35cf7b2`

File: .env.template
Comment (richbeales) [inline .env.template:1]: deleted?

**Comment by merwanehamadi** [comment]
I am going to rebase this

**Comment by merwanehamadi** [comment]
@jonathansheets517 Great summary ! so now level master is beating level A20! I am going to add your changes in:  https://github.com/Significant-Gravitas/Auto-GPT/pull/3425

because there are some conflicts in the current PR. Closing this

**Pull request closed without merging**